### PR TITLE
refactor: report renaming

### PR DESCRIPTION
**Pull request by smoralesd**
#### Description of changes

Renaming:
 - files (including test files)
 - class names (plain classes and component)
 - test names (the string we pass to `describe`)

**Notes** sadly, git was not too good handling snapshot files "renaming" (double quotes because this are autogenerated) so you'll see both new snap file and deleted snap file.

#### Pull request checklist

- [x] Addresses an existing issue: part of WI # 1558756
- [x] Added relevant unit test for your changes. (`yarn test`)
   - not affected
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
   - not affected
- [x] Ran precheckin (`yarn precheckin`)
- [x] (UI changes only) Added screenshots/GIFs to description above
   - not affected
- [x] (UI changes only) Verified usability with NVDA/JAWS
   - not affected

### CODE REVIEW

File: src/DetailsView/reports/report-generator-provider.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export const createReportGeneratorProvider = (
     assessmentReportHtmlGenerator: AssessmentReportHtmlGenerator,
 ): ReportGeneratorProvider => {
     const getGenerator = () => {
-        return new ReportGeneratorV2(reportNameGenerator, newReportHtmlGenerator, assessmentReportHtmlGenerator);
+        return new ReportGeneratorImpl(reportNameGenerator, newReportHtmlGenerator, assessmentReportHtmlGenerator);
```
Comment (dbjorge) [inline src/DetailsView/reports/report-generator-provider.ts]: Is the `ReportGeneratorProvider` still providing any value now that there's only one type of generator that might be provided?

File: src/DetailsView/reports/report-generator-impl.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ import { ReportGenerator } from './report-generator';
 import { ReportHtmlGenerator } from './report-html-generator';
 import { ReportNameGenerator } from './report-name-generator';
 
-export class ReportGeneratorV2 implements ReportGenerator {
+export class ReportGeneratorImpl implements ReportGenerator {
```
Comment (dbjorge) [inline src/DetailsView/reports/report-generator-impl.ts]: Is separating the Impl and the header interface providing any value now that there's only one impl?

File: src/DetailsView/reports/report-generator-provider.ts (hunk 1 above)
Comment (smoralesd) [reply src/DetailsView/reports/report-generator-provider.ts]: Updated. Directly impacts the `DetailsViewInitializer`, `IssuesTable` and `DetailsViewCommandBar`.

File: src/DetailsView/reports/report-generator-impl.ts (hunk 2 above)
Comment (smoralesd) [reply src/DetailsView/reports/report-generator-impl.ts]: Updated. Directly impacts the `DetailsViewInitializer`, `IssuesTable` and `DetailsViewCommandBar`.

File: src/DetailsView/details-view-initializer.ts [hunk 3]
```diff
@@ -194,7 +194,7 @@ if (isNaN(tabId) === false) {
 
             const fixInstructionProcessor = new FixInstructionProcessor();
 
-            const reportHtmlGeneratorV2 = new ReportHtmlGeneratorV2(
+            const reportHtmlGeneratorV2 = new ReportHtmlGeneratorImpl(
```
Comment (dbjorge) [inline src/DetailsView/details-view-initializer.ts]: Variable name update?

Comment (smoralesd) [reply src/DetailsView/details-view-initializer.ts]: Good catch. Though I got it before.
Updated.

**Pull request merged**